ansi/writer_test.go

```diff
@@ -61,12 +65,12 @@ func BenchmarkWriter_Write(b *testing.B) {
 		n   int
 		err error
 	)
-	b.ReportAllocs()
-	b.ResetTimer()
 
-	for i := 0; i < b.N; i++ {
-		n, err = w.Write(buf)
-	}
+	b.RunParallel(func(pb *testing.PB) {
+		for pb.Next() {
+			n, err = w.Write(buf)
```

Comment:
I'm not sure it's a good idea to reuse the same writer here. At least it's not intended/designed to be written to concurrently. We should probably also handle the error directly here. Can't we just ignore `n`?

Reply:
I think you are right about the concurrency part. And for the `n` and `err`, we can't handle them in the loop, but we can make sure they are expected after the loop.

Reply:
Also possible that I'm misunderstanding `RunParallel`. I was under the assumption that the parallel runs would overwrite each others `n` and `err`, so you would only see the results of the latest run and would miss any potential errors in other runs.

Reply:
Your understanding is totally right. But I usually cover this case in a `Test` case not `Bench` case.

Reply:
That's cool, but in that case lets omit the assignments to keep it as simple as possible :smiley:

Reply:
Do you mean no assertion in this `Bench` case?

Reply:
No, I meant: why even assign `n` and `err` if we never really use them?

We can drop these variables, unless the compiler optimizes the for-loop away entirely in such cases. I'd assume it's clever enough to realize it shouldn't in a benchmark, though.

Checking the `err` of only one of the parallel tests seems weird. Either we check all of them or none (which I'm fine with, as they're covered by separate test cases).

Reply:
Okey, I'll simplify them.